### PR TITLE
fix: allow unsetting storage secrets

### PR DESCRIPTION
**Pull request by Panaetius**
/deploy #notest

### CODE REVIEW

**Comment by Panaetius** [comment]
This change is part of the following stack:

- #415 ◀

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

**Comment by leafty** [comment]
Note: This is also implemented for data connectors secrets in https://github.com/SwissDataScienceCenter/renku-data-services/pull/413.

File: components/renku_data_services/storage/api.spec.yaml [hunk 1]
```diff
@@ -515,7 +515,9 @@ components:
           minLength: 1
           maxLength: 99
         value:
-          $ref: "#/components/schemas/SecretValue"
+          oneOf:
+            - $ref: "#/components/schemas/SecretValue"
+            - type: 'null'
```
Comment (leafty) [inline components/renku_data_services/storage/api.spec.yaml:520]: This is not allowed with the version of OpenAPI we are using:

```
"schemaValidationMessages":[{"level":"error","domain":"validation","keyword":"oneOf","message":"instance failed to match exactly one schema (matched 0 out of 2)","schema":{"loadingURI":"#","pointer":"/definitions/Components/properties/schemas/patternProperties/^[a-zA-Z0-9\\.\\-_]+$"},"instance":{"pointer":"/components/schemas/CloudStorageSecretPost"}}]
```

See how it is done for `data connctors`:
https://github.com/SwissDataScienceCenter/renku-data-services/blob/f3a5f98ef4c29baa147ab2cd6d0c52fb23148a3f/components/renku_data_services/data_connectors/api.spec.yaml#L535-L540

**Review by leafty — CHANGES_REQUESTED** [review]
Code change to `db.py` is good but the schema change is invalid:

**Review by leafty — APPROVED** [review]
LGTM

**Pull request merged**